src/content/components/Tooltip.js

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import React, { PureComponent } from 'react';
+import ReactDOM from 'react-dom';
+import type { CssPixels } from '../../common/types/units';
+
+require('./Tooltip.css');
+
+const MOUSE_OFFSET = 15;
+
+type Props = {
+  mouseX: CssPixels,
+  mouseY: CssPixels,
+  children?: React$Element<*>,
+}
+
+export default class Tooltip extends PureComponent {
+  props: Props;
+
+  state: {
+    interiorElement: HTMLElement | null,
+    interiorElementRAF: HTMLElement | null,
+  };
+
+  _isMounted: boolean;
+  _mountElement: ?HTMLElement;
+
+  constructor(props: Props) {
+    super(props);
+    (this: any)._getMountElement = this._getMountElement.bind(this);
+    this.state = {
+      interiorElement: null,
+      interiorElementRAF: null,
+    };
+  }
+
+  _getMountElement(el: HTMLElement) {
+    this.setState({ interiorElement: el });
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+    // Create a DOM node outside of the normal heirarchy.
+    const el = document.createElement('div');
+    el.className = 'tooltipMount';
```

Comment:
Having it dynamic means we don't have to worry about treading on someone else's Tooltip. I don't think practically that would happen, but I was still concerned about a collision with 2 tooltip instances.